lms/djangoapps/ccx/tests/test_models.py

```diff
@@ -125,3 +131,160 @@ def test_future_membership_disallows_auto_enroll(self):
+        actual = self.ccx.start
+        diff = expected - actual
+        self.assertEqual(diff.seconds, 0)
+
+    def test_ccx_start_caching(self):
+        """verify that caching the start property works to limit queries"""
+        now = datetime.now(UTC())
+        self.set_ccx_override('start', now)
+        with check_mongo_calls(1):
+            self.ccx.start
+        with check_mongo_calls(0):
+            self.ccx.start
+
+    def test_ccx_due_without_override(self):
+        """verify that due returns None when the field has not been set"""
+        expected = None
+        actual = self.ccx.due
+        self.assertTrue(expected is actual)
+
+    def test_ccx_due_is_correct(self):
+        """verify that the due datetime for a ccx is correctly retrieved"""
+        expected = datetime.now(UTC())
+        self.set_ccx_override('due', expected)
+        actual = self.ccx.due
+        diff = expected - actual
+        self.assertEqual(diff.seconds, 0)
+
+    def test_ccx_due_caching(self):
+        """verify that caching the due property works to limit queries"""
+        expected = datetime.now(UTC())
+        self.set_ccx_override('due', expected)
+        with check_mongo_calls(1):
+            self.ccx.due
+        with check_mongo_calls(0):
+            self.ccx.due
+
+    def test_ccx_has_started(self):
+        """verify that a ccx marked as starting yesterday has started"""
+        now = datetime.now(UTC())
+        delta = timedelta(1)
+        then = now - delta
+        self.set_ccx_override('start', then)
+        self.assertTrue(self.ccx.has_started())
+
+    def test_ccx_has_not_started(self):
+        """verify that a ccx marked as starting tomorrow has not started"""
+        now = datetime.now(UTC())
+        delta = timedelta(1)
+        then = now + delta
+        self.set_ccx_override('start', then)
+        self.assertFalse(self.ccx.has_started())
+
+    def test_ccx_has_ended(self):
+        """verify that a ccx that has a due date in the past has ended"""
+        now = datetime.now(UTC())
+        delta = timedelta(1)
+        then = now - delta
+        self.set_ccx_override('due', then)
+        self.assertTrue(self.ccx.has_ended())
+
+    def test_ccx_has_not_ended(self):
+        """verify that a ccx that has a due date in the future has not eneded
+        """
+        now = datetime.now(UTC())
+        delta = timedelta(1)
+        then = now + delta
+        self.set_ccx_override('due', then)
+        self.assertFalse(self.ccx.has_ended())
+
+    def test_ccx_without_due_date_has_not_ended(self):
+        """verify that a ccx without a due date has not ended"""
+        self.assertFalse(self.ccx.has_ended())
+
+    def test_start_datetime_short_date(self):
+        """verify that the start date for a ccx formats properly by default"""
+        start = datetime(2015, 1, 1, 12, 0, 0, tzinfo=UTC())
+        # This relies on SHORT_DATE remaining the same, is there a better way?
```

Comment:
Either confirm that tests always run with a fixed SHORT_DATE, or patch SHORT_DATE here so that you know what it is.  The first seems the better option. There are likely other tests with this reliance.